### PR TITLE
Better Handling for Blob Batch Tests in Playback

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR updates the Blob Batch tests where some operations fail, but not all the operations fail. The underlying request generation system is non-deterministic as it is able to generated multiple operations at once which may result in a different operation order each time the test is ran. This work properly when running against the live service as the request and response are tied together using a `Content-ID` marker but running the tests in playback this is an issue as it isn't capable of handling variance in ordering.

Tests where some operations are expected to fail were updated to use an order invariant check to determine correctness. All responses are checked for either having the correct response code or throwing the expected exception, they are then aggregated to validate that the number of successful operations matched what was expected. When tests are running against the live service the tests will continue to check that the operation succeeded or failed as expected in a strict ordering.

### CODE REVIEW

**Comment by alzimmermsft** [comment]
@gapra-msft @rickle-msft, could you take a look at this PR, should resolve the flaky Blob Batch test failures happening during PR validation.

File: sdk/storage/azure-storage-blob-batch/src/test/java/com/azure/storage/blob/batch/BatchAPITest.groovy [hunk 1]
```diff
@@ -128,14 +152,19 @@ class BatchAPITest extends APISpec {
...
+        notThrown(BlobBatchStorageException)
 
-        when:
-        response2.getStatusCode()
-
-        then:
-        thrown(BlobStorageException)
+        // In PLAYBACK check responses in an order invariant fashion.
+        if (testMode == TestMode.PLAYBACK) {
+            assertExpectedOrException(response1, 200) || assertExpectedOrException(response2, 200)
```
Comment (sima-zhu) [inline sdk/storage/azure-storage-blob-batch/src/test/java/com/azure/storage/blob/batch/BatchAPITest.groovy:159]: If there is definitely one will fail, it is better to check fail at &.
```
!assertExpectedOrException(response1, 200) && assertExpectedOrException(response2, 200)
```

Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob-batch/src/test/java/com/azure/storage/blob/batch/BatchAPITest.groovy:159]: The code would need to be the following, which is equivalent to what is already being used.

```java
!(assert... && assert...)
```

To be more pedantic on validation I could explicitly check each of these for being `true` and asserting the count of trues to be one. Another option is having these return `0` for failure and `1` for success and adding and checking it equals `1`.

Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob-batch/src/test/java/com/azure/storage/blob/batch/BatchAPITest.groovy:159]: Updated to doing a count validation on successful responses.

**Pull request merged**